File: genresume.py
#!/usr/bin/env python

class_names = [];

text = open('resume.txt', 'r').read()
lines = text.split("\n");

html = '<style>%s</style>' % open('style.css', 'r').read()

html += '<div id="wrapper"><table>'

def handleLine(line):
	global html, class_names
	
	if line.startswith('!'):
		class_names = [s.strip() for s in line[1:].split('\\')]
		return
	
	html += '<tr>'
	
	html += ''.join(['<td class="%s">%s</td>' % (class_names[i], processText(s)) for i, s in enumerate(line.split('\\'))])
	
	html += '</tr>'

def processText(text):
	if text.startswith('#'):
		return '<h1>%s</h1>' % text[1:]
	if text.startswith('> '):
		return '<blockquote>%s</blockquote>' % text[2:]
	return text

for line in lines: handleLine(line)

html += '</table></div>'

print(html)

open('index.html', 'w').write(html)

print('done')

File: genresume.rb
#!/usr/bin/env ruby

text = File.read 'resume.txt'
css  = File.read 'style.css'

lines = text.lines.map(&:strip).filter {|line| not line.empty?}

header, *rest = lines

classes = header[1..-1].split('\\')

trs = rest.map { |line|

	parts = line.split('\\').each_with_index.map { |content, i|

		content.strip!

		content = "<h1>#{$1}</h1>"                 if content.match /#(.+)/
		content = "<blockquote>#{$1}</blockquote>" if content.match /^> (.+)/

		"<td class=\"#{classes[i]}\">#{content}</td>"

	}

	parts = ['<td></td>'] if parts.empty?

	"<tr>#{parts.join ''}</tr>"

}

File.write 'index.html', "<style>#{css}</style><table>#{trs.join ''}</table>"